data-preprocessing/JSON_Section_Extractor/json_section_extractor.py

```diff
@@ -0,0 +1,130 @@
+import json
+import os
+import sys
+import argparse
+
+# reads bookPaths.json to obtain absolute path of the location of json files to be used.
+
+def read_PATH_json():
+    with open('booksPath.json', 'r') as file:
+        return json.load(file)
+
+# search for json files in the directory and add their names to a list
+def put_json_filenames_in_list(parsed_books_path):
+    json_files = [] 
+    for filename in os.listdir(parsed_books_path):
+        if filename.endswith(".json"):
+            json_files.append(filename)
+    return json_files
+
+# load all json file
+def load_json_file(json_file, parsed_books_path):
+    with open(parsed_books_path+json_file, "r") as file:
+        entire_json = json.load(file)
+        return entire_json[0]
+
+# obtain the section from json object (dictionary).
+def json_section(json_dictionary):
+    try:
+        section=json_dictionary["Section"]
+        return section
+    except:
+        return ''
+
+
+# nested_list argument is optional as it has a default false value. The argument is used to distinguish the two json formats used. Thus it handles the json files based on whether or not they contain nested lists or direct objects in a single list.
+def json_to_section(json_dict, pages, sections, nested_list=False): 
+    if nested_list:
+        for page in json_dict:
+            try:
+                for subpage in json_dict[page][0][1]:
+                    section = json_section(subpage)
+                    pages.append(page)
+                    sections.append(str(" ".join(section)))
+            except:
+                continue
+    else:
+        for page in json_dict:
+            try:
+                for subpage in json_dict[page]:
+                    section = json_section(subpage)
+                    pages.append(page)
+                    sections.append(str(" ".join(section)))
+            except:
+                continue
+
+
+def create_directory(directory):
+    if not os.path.exists(directory):
+        os.makedirs(directory)
+    else:
```

Comment:
is the else case needed here? seems if the if block is not executed nothing will happen so should be able to remove the else block here

Reply:
Yes please